staging/src/k8s.io/client-go/listers/discovery/v1alpha1/endpointslice_expansion.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+	"reflect"
+
+	discovery "k8s.io/api/discovery/v1alpha1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+)
+
+// EndpointSliceListerExpansion allows custom methods to be added to
+// EndpointSliceLister.
+type EndpointSliceListerExpansion interface {
+	GetMatchingSubset(endpointSliceKey string, labelKeys []string) ([]discovery.Endpoint, error)
+	GetMatchingServiceSubset(endpointSliceKey string) ([]discovery.Endpoint, error)
```

Comment:
These interfaces aren't intended to be places to put "helper" type methods.

The two cases we've used them for are:
* adding methods to address subresource-specific methods the generated client did not understand (like namespace finalize)
* adding methods to address resource methods that the generated client does not understand (like tokenreview create)

Reply:
Thanks for the clarification, I'd thought this was a fairly similar use case to the ServiceListerExpansion (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/listers/core/v1/service_expansion.go#L36). Is there a good place to put these kinds of helper functions?

Reply:
https://github.com/kubernetes/kubernetes/blob/0c0408c790c64d9105a85ae702c73d0714bdb850/staging/src/k8s.io/client-go/listers/core/v1/service_expansion.go#L34-L35

:)

Depends on the target audience. For controllers inside kubernetes/kubernetes, pkg/controller/util or pkg/api/v1 tend to attract resource specific subpackages with helper methods.

Reply:
haha, woops, thanks! This was more intended to make EndpointSlices easier to use for potential consumers, so not completely sure where it will end up making the most sense. We do already have an endpoint controller util, but up until now that's primarily been used to share common functions between the endpoint and endpointslice controller. This may make the most sense in a new `pkg/api/endpointslice` util package. Thanks for the help!